bookdown/07-special-density.rmd

```diff
@@ -0,0 +1,89 @@
+## Density Estimation {#density}
+
+Density estimation is the learning task to find the unknown distribution from which an i.i.d. data set is generated. We interpret this broadly, with this distribution not necessarily being continuous (so may possess a mass not density). The conditional case, where a distribution is predicted conditional on covariates, is known as ‘probabilistic supervised regression’, and will be implemented in `r mlr_pkg("mlr3proba")` in the near-future. Unconditional density estimation is viewed as an unsupervised task.
+
+The package `r mlr_pkg("mlr3proba")` extends `r mlr_pkg("mlr3")` with the following objects for density estimation:
+
+* `r ref("mlr3proba::TaskDens", text = "TaskDens")` to define density tasks
+* `r ref("mlr3proba::LearnerDens", text = "LearnerDens")` as base class for density estimators
+* `r ref("mlr3proba::PredictionDens", text = "PredictionDens")` as specialized class for `r ref("Prediction")` objects
+* `r ref("mlr3proba::MeasureDens", text = "MeasureDens")` as specialized class for performance measures
+
+In this example we demonstrate the basic functionality of the package on the `r ref("datasets::faithful", text = "faithful")` data from the `r cran_pkg("datasets")` package.
+This task ships as pre-defined `r ref("TaskDens")` with `r mlr_pkg("mlr3proba")`.
+
+```{r 07-special-density-001}
+library("mlr3")
+library("mlr3proba")
+
+task = tsk("precip")
+print(task)
+
+# the target column is a numeric vector:
+head(task$truth())
+
+# histogram and density plot
+library("mlr3viz")
+autoplot(task, type = "overlay")
+```
+
+Unconditional density estimation is an unsupervised method. Hence, `TaskDens` is an unsupervised task which inherits directly from `Task` unlike `TaskClassif` and `TaskRegr`. However, `TaskDens` still has a `target` and a `$truth` field defined by:
+  
+  * `target` - the variable for which to estimate density
+  * `truth` -  the `target`. This is *not* the true density which is always unknown. 
```

Comment:
`truth` - the values of the target column.

?

Reply:
`truth` -  the values of the `target` column (which is *not* the true density which is always unknown)

?

Reply:
Yes, this is what I meant. I just wanted it to include the word "values".